editor-footnotes.php

```diff
@@ -52,3 +52,18 @@ function autoload( $cls ) {
 
 // Load the Editor class on admin pages
 add_action( 'admin_init', [ '\Editor_Footnotes\Editor', 'instance' ] );
+
+/**
+ * Allow span[data-footnote] to pass through post content.
+ *
+ * @param  array $kses Allowed HTML tags.
+ * @param  string $context Kses context.
+ * @return array
+ */
+function wp_kses_allowed_html( $kses, $context ) {
+	if ( 'post' === $context ) {
```

Comment:
In that case, we'd still want to do what we're doing here (it wouldn't cause a PHP notice if that's your concern), because otherwise this plugin wouldn't work at all.